src/wallet/wallet.cpp

```diff
@@ -4773,6 +4767,10 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
         {
             return UIError(strprintf(_("Wallet needed to be rewritten: restart %s to complete"), _(PACKAGE_NAME)));
         }
+        else if (nLoadWalletRet == DB_WRONG_NETWORK)
+        {
+            return UIError(strprintf(_("Wallet %s is from a different network (mainnet, testnet, regtest)"), walletFile));
```

Comment:
Hmm, I don't like that the error message is less informative, but I can deal with that for now. However, listing all three networks here is confusing, and also misleading (it could be failing due to being a wallet from a code- or chain-fork).
```suggestion
            return UIError(strprintf(_("Wallet %s is not for %s %s network"), walletFile, _(PACKAGE_NAME), params.NetworkIDString()));
```